src/App.tsx

```diff
@@ -1,14 +1,77 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useCallback, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+function getFilteredTodos(key: string, query: string, todosToFilter: Todo[]) {
+  let filteredTodos = [];
+
+  switch (key) {
+    case 'all': {
+      filteredTodos = todosToFilter;
+      break;
+    }
+
+    case 'active': {
+      filteredTodos = todosToFilter.filter(todo => !todo.completed);
+      break;
+    }
+
+    case 'completed': {
+      filteredTodos = todosToFilter.filter(todo => todo.completed);
+      break;
+    }
+
+    default:
+      filteredTodos = todosToFilter;
+  }
+
+  filteredTodos = filteredTodos.filter(todo => todo.title.toLowerCase().includes(query.toLowerCase()));
+
+  return filteredTodos;
+}
 
 export const App: React.FC = () => {
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loading, setLoading] = useState(false);
```

Comment:
You are using a boolean value for this var, so consider naming it something like `isLoading`